save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestSuitesSourceController.kt

```diff
@@ -352,6 +349,36 @@ class TestSuitesSourceController(
         }
         .map { it.toDto() }
 
+    @PostMapping("/api/$v1/test-suites-sources/{organizationName}/get-or-create")
+    @RequiresAuthorizationSourceHeader
+    @PreAuthorize("permitAll()")
+    @Operation(
+        method = "POST",
+        summary = "Get or create a new test suite source by provided values.",
+        description = "Get or create a new test suite source by provided values.",
+    )
+    @Parameters(
+        Parameter(name = "organizationName", `in` = ParameterIn.QUERY, description = "name of organization", required = true),
+    )
+    @ApiResponse(responseCode = "200", description = "Successfully get or create test suites source with requested values.")
+    @ApiResponse(responseCode = "409", description = "Organization was not found by provided name.")
+    @ApiResponse(responseCode = "409", description = "Git credentials was not found by provided url.")
+    fun createTestSuitesSource(
+        @PathVariable organizationName: String,
+        @RequestBody testSuiteRequest: TestSuitesSourceDto,
+    ): Mono<StringResponse> = getOrganization(organizationName)
+        .zipWhen { organization ->
+            gitService.findByOrganizationAndUrl(organization, testSuiteRequest.gitDto.url)
+                .toMono()
+                .switchIfEmptyToResponseException(HttpStatus.CONFLICT) {
+                    "There is no git credential with url ${testSuiteRequest.gitDto.url} in $organizationName"
+                }
+        }
+        .map { (organization, git) ->
+            testSuitesSourceService.getOrCreate(organization, git, testSuiteRequest.testRootPath, testSuiteRequest.branch)
```

Comment:
can we create a separate method to create testSuitesSource by DTO entity?
current method `getOrCreate` generates `name` automatically as url to github. User can input something specific

Reply:
Sure, will do it

Reply:
Done

Reply:
Yep, but now there is a method `fetch` in testSuitesSourceService